pkgs/applications/graphics/sane/backends/brscan4/preload.c

```diff
@@ -0,0 +1,170 @@
+/* Brgen4 search for configuration under `/etc/opt/brother/scanner/brscan4`. This
```

Comment:
@dezgeg I looked at the `libredirect.c` solution. It is not generic enough for what I do here.

The simpler things it is missing:

 -  execvp
 -  opendir

Those could definitely be added to the generic library.

The harder things it is missing are:

 -  system

    This is a command string (i.e.: this is not a path) which contain a path not at its
    beginning. My implementation is currently limited to this case (only replace the
    first found path and maximal buffer may not be adapted to all cases).

 -  The brscan4 tool is writing to a file. I use the `BRSANENETDEVICE4_CFG_FILENAME`
    env var to control where it is written. I cant see how to integrate this to the generic tool.

In order to add such case through the generic path, the generic path should allow for
a way to add another layer of custom preload (is that even possible?).